addons/missile_manpad/functions/fnc_eachFrame.sqf

```diff
@@ -0,0 +1,127 @@
+/*
+ * Author: PabstMirror
+ * Each frame.
+ *
+ * Arguments:
+ * 0: Args <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [[]] call ace_missile_manpad_fnc_eachFrame
+ *
+ * Public: No
+ */
+params ["_args"];
+_args params ["_unit", "_pfid", "_actionId", "_config", "_lockCanidate", "_lockStartTime", "_haveLock", "_lastSound", "_lastSoundTimeout"];
+_config params ["_seekerMaxRange", "_lockAngle", "_uncageAngle", "_lockingTimeMin", "_lockingTimeMax", "_lockingSound", "_lockedSound"];
+
+private _fnc_playSound = {
+    params ["_sound"];
+    if ((_lastSound != _sound) || {CBA_missionTime > _lastSoundTimeout}) then {
+        private _affected = (ASLToAGL eyePos _unit) nearEntities ["CAManBase", 50];
+        [QGVAR(soundEffect), [_unit, _sound], _affected] call CBA_fnc_targetEvent;
+        _args set [7, _sound];
+        _args set [8, CBA_missionTime + 10]; // tiny clipping when re-starting track
+    };
+};
+
+private _fnc_searchTarget = {
+    params ["_target", "_currentAngleMax"];
+
+    private _aimASL = aimPos _target;
+    private _targetDiff = _aimASL vectorDiff _seekerASL;
+    private _angle = acos ((_seekerDir vectorCos _targetDiff) min 1 max -1); // vectorCos will sometimes be 1.0001 :(
+    private _visibility = [_source, "VIEW", _target] checkVisibility [_seekerASL, _aimASL];
+    private _dist = vectorMagnitude _targetDiff;
+
+    if ((_angle > _currentAngleMax) || {_visibility < 0.1} || {_dist > _seekerMaxRange}) exitWith {
+        if (EGVAR(missileguidance,debug_drawGuidanceInfo)) then {
+            drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Cursors\select_target_ca.paa", [1,0,0,1],
+                ASLToAGL _aimASL, 1.5, 1.5, 45, format ["a%1 - v%2 - d%3", _angle, _visibility, _dist], 0.5, 0.025, "TahomaB"];
+        };
+        0
+    };
+
+    private _ret = _visibility // get a signal strength estimate
+        * linearConversion [_currentAngleMax/2, _currentAngleMax, _angle, 1, 0.5, true]
+        * linearConversion [_seekerMaxRange/2, _seekerMaxRange, vectorMagnitude _targetDiff, 1, 0.5, true]
+        * linearConversion [100, 3000, getMass _target, 0.25, 1, true];
+
+    if (EGVAR(missileguidance,debug_drawGuidanceInfo)) then {
+        drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Cursors\select_target_ca.paa", [0,1,0,1],
+            ASLToAGL _aimASL, 1.5, 1.5, 45, format ["a%1 - v%2 - d%3 = %4", _angle, _visibility, _dist, _ret], 0.5, 0.025, "TahomaB"];
+    };
+    _ret
+};
+
+
+if (!GVAR(isLockKeyDown) || {(currentMagazine _unit) == ""}) exitWith {
+    "" call _fnc_playSound;
+    _unit setVariable [QEGVAR(missileguidance,target), nil];
+    _args set [4, objNull]; // _lockCanidate = objNull
+};
+
+private _source = _unit;
+private _seekerASL = eyePos _source;
+private _seekerDir = _source weaponDirection currentWeapon _source;
+
+private _lockFeedback = 0;
+if (isNull _lockCanidate) then {
+    // find any target within seeker range
+    private _potentialTargets = _source nearEntities ["Air", _seekerMaxRange];
```

Comment:
flares won't be locked with this, idk if thats desired

Reply:
I was wondering if this should more closely just copy the ir_seeker code

Reply:
i think it should; i'd want to break the logic into it's own function just because its a but unweidly and if it changes in the future it'd be a pain to update twice